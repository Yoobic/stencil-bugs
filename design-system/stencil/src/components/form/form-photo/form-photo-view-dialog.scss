:host {
    .images {
        position: relative;
        min-height: 300px;
        text-align: center;
    }
    .preview {
        width: 100%;
        &.edit {
            @include abs-pos(0, auto, auto, 0);
            z-index: 1;
        }
        &.web {
            width: auto;
            max-width: 100%;
            max-height: 600px;
            left: 0;
            right: 0;
        }
    }
    .image-recos {
        @include abs-pos(0.625rem, auto, auto, 0.625rem); // top: 10px;
        opacity: 0.9;
        width: 250px;
        z-index: 10;
        .image-reco {
            height: 30px;
            padding: $padding/2;
            margin-bottom: $padding/4;
            border-radius: $border-radius;
            border-width: 2px;
            background: cssvar(light);
            color: cssvar(text-header);
            .image-reco-value {
                float: right;
            }
        }
    }
}