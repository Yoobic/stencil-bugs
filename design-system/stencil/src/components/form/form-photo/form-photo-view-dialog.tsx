import { Component, Element, Prop } from '@stencil/core';
import { closeModal } from '../../../utils/helpers/helpers';

@Component({
    tag: 'yoo-form-photo-view-dialog',
    styleUrl: 'form-photo-view-dialog.scss',
    scoped: true
})
export class YooFormPhotoViewDialogComponent {

    @Prop() label: string;
    @Prop() preview: any;
    @Prop() allowAnnotate: boolean;
    @Prop() edit: any;
    @Prop() imageRecognitionResults: Array<{ title: string; value: any; color: string; hidden?: boolean; }>;
    @Prop() isStitch: boolean;

    @Element() host: HTMLStencilElement;

    // private showImageRecognitionResults: boolean = false;

    onCancel() {
        closeModal(null);
    }

    render() {
        return [
            //<div class={'shadow header ' + (services.coreConfig.isIphoneX() ? 'iphone-x' : '')}>
            <ion-header class="shadow" no-border>
                <ion-toolbar color="light">
                    <ion-buttons slot="start">
                        <ion-button class="close" color="dark" onClick={() => this.onCancel()}>
                            <i slot="icon-only" class="yo-close"></i>
                        </ion-button>
                    </ion-buttons>
                    <ion-title>{this.label}</ion-title>
                </ion-toolbar>
            </ion-header>,
            // </div>,
            //<div class="content">
            <ion-content scrollEnabled={false}>
                <div class="images">
                    {!(this.edit && this.isStitch) ? <img class="preview" src={this.preview} /> : null}
                    {this.edit ? <img class="preview edit" src={this.edit} /> : null}
                    {this.imageRecognitionResults && this.imageRecognitionResults.length > 0 ?
                        <div class="image-recos">
                            {this.imageRecognitionResults.map((result) => {
                                return !result.hidden ? <div class="image-reco"><span class="image-reco-value">{result.title}:{result.value}</span></div> : null;
                            })}
                        </div> : null}
                </div>
            </ion-content>
            // </div>
        ];
    }
}