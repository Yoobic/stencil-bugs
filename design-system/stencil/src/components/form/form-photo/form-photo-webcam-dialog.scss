:host {
    ion-content {
      /deep/  #webcamHolder {
            width: 100%;
            height: 100%;
        }
        /deep/ .scroll-inner {
            height: 100%;
        }
    }
}