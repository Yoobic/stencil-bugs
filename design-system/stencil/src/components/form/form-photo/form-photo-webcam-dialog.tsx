import { Component, Element, Prop } from '@stencil/core';
import { services } from '../../../services';
import { closeModal } from '../../../utils/helpers/helpers';
import * as JpegCamera from 'jpeg-camera/dist/jpeg_camera_no_flash';

@Component({
    tag: 'yoo-form-photo-webcam-dialog',
    styleUrl: 'form-photo-webcam-dialog.scss',
    scoped: true
})
export class YooFormPhotoWebcamDialogComponent {

    @Prop() device: any;

    @Element() host: HTMLStencilElement;

    private holderRef: HTMLElement;
    private webcamCamera;
    private _jpegCamera = (window as any).JpegCamera || JpegCamera;

    componentDidLoad() {
        this.holderRef = this.host.querySelector('#webcamHolder');
        this.initWebcam(this.device.deviceId);
    }

    componentDidUnload() {
        this.cleanUpCamera();
    }

    initWebcam(deviceId: string) {
        //this.getJpegCamera().then(() => {
        if (this.webcamCamera) {
            this.webcamCamera.stop();
        }
        let options: any = {
            shutter_ogg_url: 'assets/audio/shutter.ogg',
            shutter_mp3_url: 'assets/audio/shutter.mp3'
        };

        if (this.device) {
            options.deviceId = this.device.deviceId;
            options.mirror = false;
        }
        if (this.holderRef) {
            try {
                this.webcamCamera = new this._jpegCamera(this.holderRef, options);
            } catch (e) { }
        }
        //});
    }

    cleanUpCamera() {
        if (this.webcamCamera) {
            try {
                if (this.webcamCamera && this.webcamCamera.audio_context && this.webcamCamera.audio_context.state !== 'closed') {
                    this.webcamCamera.audio_context.close();
                }
            } catch (err) { }
            try {
                this.webcamCamera.stop();
            } catch (err) { }
        }
    }

    onCancel() {
        this.cleanUpCamera();
        closeModal(null);
    }

    onSave() {
        if (this.webcamCamera) {
            try {
                let snapshot = this.webcamCamera.capture();
                snapshot.get_canvas(canvas => {
                    let canvasData = canvas.toDataURL('image/jpeg', 0.7);
                    //this.cleanUpCamera();
                    if (canvasData && canvasData.length > 10) {
                        closeModal(canvasData);
                    }
                });
            } catch (err) { }
        }
    }

    render() {
        return [
            //<div class={'shadow header ' + (services.coreConfig.isIphoneX() ? 'iphone-x' : '')}>
            <ion-header class="shadow" no-border>
                <ion-toolbar color="light">
                    <ion-buttons slot="start">
                        <ion-button class="close" color="dark" onClick={() => this.onCancel()}>
                            <i slot="icon-only" class="yo-close"></i>
                        </ion-button>
                    </ion-buttons>
                    <ion-title>{services.translate.get('ADVANCED')}</ion-title>
                    <ion-buttons slot="end" onClick={() => this.onSave()}>
                        <ion-button color="success">{services.translate.get('SAVE')}</ion-button>
                    </ion-buttons>
                </ion-toolbar>
            </ion-header>,
            //</div>,
            //<div class="content">
            <ion-content scrollEnabled={false} class="bg-light">
                <div id="webcamHolder">Webcam</div>
            </ion-content>
            //</div>
        ];
    }
}