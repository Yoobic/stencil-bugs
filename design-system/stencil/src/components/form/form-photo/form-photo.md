---
name: Photo
category: Form Components
---

```yoo-form-photo.html
    <yoo-form-photo></yoo-form-photo>

    <h3>With a value</h3>
    <yoo-form-photo value="linkPhoto"></yoo-form-photo>
```

### Required
```yoo-form-photo-required.html
    <h3>Required</h3>
    <yoo-form-photo required="true"></yoo-form-photo>
```

### Read Only
```yoo-form-photo-readonly.html
    <h3>Read Only</h3>
    <yoo-form-photo readonly="true" value="linkPhoto"></yoo-form-photo>
```

## Interface
```tsx
export declare class IAlgorithm extends IEntity {
    channel: string;
    name: string;
    server: string;
}
```

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`value`| `string or Array<string>` | `none` | mutable prop, the value of the form |
|`extraData`| `any` | `{}` |  |
|`validators`| `Array<Validator<string> or ValidatorEntry>` | `[]` | `` |
|`asyncValidators `| `Array<AsyncValidator<string>>` | `` | `` |
|`placeholder`| `string` | `none` | the placeholder text |
|`required`| `boolean` | `false` | if the form is required or not |
|`readonly`| `boolean` | `false` | render as readonly, `false` to render as editable |
|`type`| `'video' or 'audio' or 'photo'` | `` |  the type of file to capture|
|`multiple`|`boolean`| `false` |if you can pick different value or not|
|`min`|`number`|`1`|minimum number of file to upload|
|`max`|`number`|`1`|maximum number of file to upload|
|`maxWidth`|`number`|`none`| the maximum width of photo|
|`duration`|`number`|`none`|duration of the capture video / audio|
|`saveGeoloc`|`boolean`|`false`|whether it saves the location in the picture taken or not|
|`allowLibrary`|`boolean`|`false`|whether it allows to import from librabry or not|
|`allowAnnotate`|`boolean`|`false`|whether it allows to annotate to the photo view dialog or not|
|`isImageRecognition`|`boolean`|`false`|whether it show the button image recognition or not|
|`algorithm`|`IAlgorithm`|`none`|the algorithm used for the image recognition|
|`isBackgroundProcess`|`boolean`|`false`|hide the recognition button when `true` |
|`label`|`string`|`none`| the value of the photo preview label|

## Methods
|Name|Parameters|Description|
|---|---|---|
|`isValid`|`none`| return validity (`boolean`)|

## Events
|Attr|Type|Description|
|---|---|---|
|`validityChanged`|`boolean`|event emitted when the validity changes|
|`inputBlurred`|`any`|event emitted when the input is blurred|
|`inputFocused`|`boolean`|event emitted when the form input is focused|
|`inputChanged`|`any`|event emitted when the form input changes|
|`imageRecognition`|`any`|event emitted when clicking on the recognition button|