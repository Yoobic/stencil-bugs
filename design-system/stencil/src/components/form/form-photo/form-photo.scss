:host {
    .container {
        align-items: start;
        padding: 0.5rem 0rem;
        yoo-progress-bar {
            padding-right: 0.5rem;
            margin-bottom: 0.5rem;
        }
        .camera-container {
            position: relative;
            border: 1px solid cssvar(stable-30);
            border-radius: 0.5rem;
            width: 105px;
            height: 105px;
            cursor: pointer;
            .required {
                font-size: 15px;
                color: cssvar(success);
                @include abs-pos(auto, auto, auto, 0.625rem);
                
                
            }
            i {
                @include abs-pos(2.1875rem, auto, auto, 2.1875rem);
                
                // top: 35px;
                // left: 35px;
                font-size: 2rem;
            }
            .preview,
            .edit,
            .overlay {
                @include abs-pos(0, auto, auto, auto);
                
                
                border-radius: 0.5rem;
                width: 105px;
                height: 105px;
            }
            .overlay {
                opacity: 0.5;
                background: cssvar(dark-60)
            }
            input[type=file] {
                @include abs-pos(0, auto, auto, 50%);
                
                
                
                display: block;
                width: 105px;
                height: 105px;
                border-radius: 50%;
                margin-left: -50px;
                text-align: right;
                filter: alpha(opacity=0);
                opacity: 0;
                outline: none;
                background: cssvar(light);
                cursor: inherit;
                pointer-events: none;
            }
        }
        .multiple {
            .camera-container {
                display: inline-block;
                margin-right: 0.5rem;
                margin-bottom: 0.5rem;
                .yo-plus {
                    width: 25px;
                    height: 25px;
                    font-size: 1rem;
                    top: 40px;
                    left: 40px;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    background: cssvar(success);
                    border-radius: 20px;
                    &:before {
                        color: cssvar(light);
                    }
                }
            }
        }
        .button-container {
            margin-left: 30px;
            .balanced {
                background-image: $gradient-success;
            }
        }
        button {
            font-size: 14px;
            font-weight: bold;
            color: cssvar(light);
            width: 160px;
            height: 52px;
            border-radius: 26px;
        }
    }
    .readonly {
        .preview {
            width: 100%;
            height: 250px;
            margin-bottom: 0.5rem;
            border-radius: $border-radius-input;
        }
    }
}