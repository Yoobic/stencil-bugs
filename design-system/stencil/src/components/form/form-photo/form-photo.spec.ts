import { TestWindow } from '@stencil/core/testing';
import { YooFormPhotoComponent } from './form-photo';

describe('YooFormPhotoComponent', () => {
    it('should build', () => {
        expect(new YooFormPhotoComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let window, element;
        beforeEach(async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooFormPhotoComponent],
                html: '<yoo-form-photo></yoo-form-photo>'
            });
        });

        it('Should render', async () => {
            expect(element).toMatchSnapshot();
        });

        it('Should render with a value prop', async () => {
            element.value = 'my value';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with extraData prop', async () => {
            element.extraData = { edit: null, texts: [] };
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with placeholder prop', async () => {
            element.placeholder = 'my placeholder';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render as required', async () => {
            element.required = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render as readonly', async () => {
            element.readonly = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with photo type', async () => {
            element.type = 'photo';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with video type', async () => {
            element.type = 'video';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with audio type', async () => {
            element.type = 'audio';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with multiple', async () => {
            element.multiple = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with allow library', async () => {
            element.allowLibrary = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with allow annotate', async () => {
            element.allowAnnotate = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with image recognition button', async () => {
            element.isImageRecognition = true;
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should render with label', async () => {
            element.label = 'my label';
            await window.flush();
            expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {
        let emitEvent, window;
        beforeEach(async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });

        // it('Should emit the value prop when it changes', async () => {

        // });
    });
});