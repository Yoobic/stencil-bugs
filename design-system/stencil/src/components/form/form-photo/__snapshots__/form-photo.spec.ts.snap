// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormPhotoComponent Rendering Should render 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render as readonly 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render as required 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        <span
          class="required"
          data-yoo-form-photo=""
        >
          *
        </span>
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with a value prop 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <div
          class="preview"
          data-yoo-form-photo=""
          style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%;"
        />
        
        <div
          class="overlay"
          data-yoo-form-photo=""
        />
        <i
          class="yo-camera-solid light"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with allow annotate 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with allow library 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with audio type 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="success yo-microphone-solid"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with extraData prop 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with image recognition button 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      <div
        attr-layout="row"
        attr-layout-align="center center"
        class="button-container"
        data-yoo-form-photo=""
      >
        <button
          class="button balanced"
          data-yoo-form-photo=""
          type="button"
        >
          PHOTOLIVECOUNTER
        </button>
      </div>
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with label 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with multiple 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="multiple"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <i
          class="yo-plus"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with photo type 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with placeholder prop 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="yo-camera-solid success"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;

exports[`YooFormPhotoComponent Rendering Should render with video type 1`] = `
<yoo-form-photo
  class="hydrated"
  data-yoo-form-photo-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-photo=""
  >
    
    <div
      attr-layout="row"
      class="single"
      data-yoo-form-photo=""
    >
      <div
        class="camera-container"
        data-yoo-form-photo=""
      >
        <input
          accept="accept"
          data-yoo-form-photo=""
          type="file"
        />
        
        <i
          class="success yo-videocamera-solid"
          data-yoo-form-photo=""
        />
      </div>
      
    </div>
  </div>
</yoo-form-photo>
`;
